### PR TITLE
feat: major version upgrades for frontend-platform & frontend-build

### PR DESCRIPTION
**Pull request by BilalQamar95**
### Description

- Major version upgrades, Updating `frontend-build` to `v14` & `frontend-platform` to `v8`
- Updated `snapshots` for failing tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
   },
   "dependencies": {
     "@edx/brand": "npm:@openedx/brand-openedx@^1.2.2",
-    "@edx/frontend-platform": "7.1.3",
+    "@edx/frontend-platform": "8.0.0",
```
Comment (brian-smith-tcril) [inline package.json:36]:
```suggestion
    "@edx/frontend-platform": "^8.0.0",
```

File: package.json [hunk 2]
```diff
@@ -73,7 +73,7 @@
   "devDependencies": {
     "@edx/browserslist-config": "^1.1.1",
     "@edx/reactifex": "1.1.0",
-    "@openedx/frontend-build": "13.1.4",
+    "@openedx/frontend-build": "14.0.3",
```
Comment (brian-smith-tcril) [inline package.json:76]:
```suggestion
    "@openedx/frontend-build": "^14.0.3",
```

**Review by brian-smith-tcril — COMMENTED** [review]
Sorry this has been sitting unreviewed for so long! I would love to see this merged!

I left a couple suggestions to switch to `^` for dependency pinning instead of using exact versions.

If you could rebase this to resolve the merge conflicts I'll gladly push to get this merged!

Thanks so much for this PR!

**Comment by BilalQamar95** [comment]
@brian-smith-tcril I have updated the versions with caret & resolved the conflicts

**Review by brian-smith-tcril — APPROVED** [review]
Thank you!

**Pull request merged**